### ISSUE TITLE
Unlcear that BLE001 is suppressed when logging/raising error.

### ISSUE THREAD

**Issue by ollz272** [astral-sh/ruff#9571]
The documentation isn't clear when BLE001 gets suppressed. Looking at the following code:

```python
"""TEST."""
import logging

log = logging.getLogger(__name__)


def my_func() -> None:  # noqa: D103
    pass


try:
    my_func()
except Exception:
    log.exception("hello")

try:
    my_func()
except Exception:
    my_func()
    raise

try:
    my_func()
except Exception:
    my_func()
```

Only the last try statement gets picked up as BLE001. So im guessing the following rules apply:
- If we log the exception in the except block, dont raise BLE001
- If we raise the exception in the except block, dont raise BLE001

However, this isn't clear from the documentation - it just implies bare excepts will cause it. Can this be updated so we understand the rules better, please?
